src/cargo/util/semver_ext.rs

```diff
@@ -108,6 +108,125 @@ impl From<VersionReq> for OptVersionReq {
     }
 }
 
+#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug)]
+pub struct PartialVersion {
```

Comment:
Some prior art <https://github.com/oli-obk/cargo_metadata/pull/243>.

Do you feel like `BareVersion` is a better name? I am good with either.

Reply:
For me, `Partial` better conveys than `Bare`.  If there is prior art for `Bare`, I'm fine with it.

Reply:
From [here](https://doc.rust-lang.org/nightly/cargo/reference/manifest.html#the-rust-version-field),

> The Rust version must be a bare version number with two or three components; 

Let's have a record and move on with `PartialVersion`. It's not worth too much efforts here :)